uploader.go

```diff
@@ -184,22 +192,47 @@ func (u UploadOptions) contextWithTimeout(ctx context.Context) (context.Context,
 	return context.WithTimeout(ctx, u.BatchWindow)
 }
 
+func addToPendingBatch(u UploadOptions, inputData InputData) Batch {
+	blob := inputData.GetBlob()
+	objectType := inputData.GetObjectType()
+	b, ok := pendingBatches[objectType]
+
+	// start a new batch
+	if !ok {
+		firstBlobTime := time.Now()
+		b = Batch{
+			firstBlobTime: firstBlobTime,
+			size:          uint64(len(blob)),
+			key:           inputData.GetKey(firstBlobTime),
+			objectType:    objectType,
+		}
+
+		b.buffer = <-u.bufferPool
+		pendingBatches[objectType] = b
+	} else {
+		b.size += uint64(len(blob))
+	}
+
+	b.buffer = append(b.buffer, blob)
```

Comment:
Adding a blob to a batch (appending and updating size) should be done via channel.

Reply:
How come?